server/game/core/cost/CostAdjuster.ts

```diff
@@ -1,37 +1,57 @@
 import type { AbilityContext } from '../ability/AbilityContext';
 import type { IAbilityLimit } from '../ability/AbilityLimit';
 import type { Card } from '../card/Card';
-import { CardType, PlayType, Aspect, CardTypeFilter } from '../Constants';
+import { PlayType, Aspect, CardTypeFilter, AspectFilter } from '../Constants';
 import type Game from '../Game';
 import type Player from '../Player';
 import * as EnumHelpers from '../utils/EnumHelpers';
+import * as Contract from '../../core/utils/Contract';
 
-export enum CostAdjustDirection {
+export enum CostAdjustType {
     Increase = 'increase',
-    Decrease = 'decrease'
+    Decrease = 'decrease',
+    IgnoreAllAspects = 'ignoreAllAspects',
+
+    /** @deprecated Not yet implemented: TODO with Kylo/Rey */
+    IgnoreSpecificAspects = 'ignoreSpecificAspect'
 }
 
 // TODO: refactor so we can add TContext for attachTargetCondition
 export interface ICostAdjusterProperties {
+
+    /** The type of cards that can be reduced */
     cardTypeFilter: CardTypeFilter;
-    amount: number | ((card: Card, player: Player) => number);
-    direction: CostAdjustDirection;
+
+    /** The amount to adjust the cost by */
+    amount?: number | ((card: Card, player: Player) => number);
+
+    /** The type of cost adjustment */
+    costAdjustType: CostAdjustType;
+
+    /** The cost floor (for minimum costs) */
     costFloor?: number;
+
+    /** The number of cost reductions permitted */
     limit?: IAbilityLimit;
```

Comment:
Can you tag this and `playingTypes` as @deprecated with a note that they aren't tested yet